src/compiler/config/validate-config.ts

```diff
@@ -104,24 +104,40 @@ export const validateConfig = (
   }
 
   validatedConfig.extras = validatedConfig.extras || {};
-  validatedConfig.extras.appendChildSlotFix = !!validatedConfig.extras.appendChildSlotFix;
-  validatedConfig.extras.cloneNodeFix = !!validatedConfig.extras.cloneNodeFix;
   validatedConfig.extras.lifecycleDOMEvents = !!validatedConfig.extras.lifecycleDOMEvents;
   validatedConfig.extras.scriptDataOpts = !!validatedConfig.extras.scriptDataOpts;
-  validatedConfig.extras.slotChildNodesFix = !!validatedConfig.extras.slotChildNodesFix;
   validatedConfig.extras.initializeNextTick = !!validatedConfig.extras.initializeNextTick;
   validatedConfig.extras.tagNameTransform = !!validatedConfig.extras.tagNameTransform;
-  // TODO(STENCIL-914): remove `experimentalSlotFixes` when it's the default behavior
-  validatedConfig.extras.experimentalSlotFixes = !!validatedConfig.extras.experimentalSlotFixes;
 
-  // if the user has set `extras.experimentalSlotFixes` then we turn on all of
-  // the slot-related fixes automatically.
+  // TODO(STENCIL-914): remove when `experimentalSlotFixes` is the default behavior
+  // If the user set `experimentalSlotFixes` and any individual slot fix flags, we need to log a warning
+  // to the user that we will "override" the individual flags
+  if (
+    validatedConfig.extras.experimentalSlotFixes === true &&
+    ((validatedConfig.extras as any).appendChildSlotFix === true ||
+      (validatedConfig.extras as any).cloneNodeFix === true ||
+      (validatedConfig.extras as any).slotChildNodesFix === true ||
+      (validatedConfig.extras as any).scopedSlotTextContentFix === true)
+  ) {
+    const warning = buildError(diagnostics);
+    warning.level = 'warn';
+    warning.messageText =
+      'The `experimentalSlotFixes` flag cannot be used in combination with other slot fix flags. These individual flags will be ignored. Please update your Stencil config accordingly.';
```

Comment:
I'm no wordsmith so open to better phrasing here 😂

Reply:
Based on the outcome of https://github.com/ionic-team/stencil/pull/4767/files#r1318694085, this might be moot, but one thing we could do is collect a `Set` of flags that are erroneously set:

```ts
const flagsToCollect = new Set<string>();
if (validatedConfig.extras.experimentalSlotFixes === true) {
  ['appendChildSlotFix', 'slotChildNodesFix', 'cloneNodeFix', 'scopedSlotTextContentFix'].forEach((flag) => {
    if (validatedConfig.extras[flag] === true) {
      flagsToCollect.add(flag);
    }
  });
}
```

then print them out to help folks understand what a 'slot fix flag' "means":
```ts
    warning.messageText =
      'The `experimentalSlotFixes` flag cannot be used in combination when the following flags are set: ${[...flagsToCollect].join(',')}. These individual flags will be ignored. Please update your Stencil config accordingly.';
```